### ISSUE TITLE
[Feature] Allow a rejection reason for pending accounts

### ISSUE THREAD

**Issue by joyeusenoelle** [tootsuite/mastodon#10680]
Currently there is no way for a user with a pending account to know *why* their account hasn't been approved. This feature would add an email to rejected accounts, and give the option to customize specialized responses.

This suggestion has two parts:

* The Reject button should be paired with a drop-down menu featuring configurable rejection messages. It should always contain a default rejection message, the text of which can also be configured, and which might read like this to start:

> Based on the information you provided, we don't believe you'll be a good fit for our Mastodon instance at `$INSTANCE_NAME`. Please use the Mastodon instance picker at https://instances.social to find an instance that might be more suitable for you.

This should be configurable in the admin panel.

* The admin (and possibly moderators) should be able to add custom rejection messages, with a title and custom text. The title is what will display in the rejection drop-down menu (and should *not* be shown to the user); the custom text is what will be emailed to the rejected user.

For example, on mastodon.social, we get many people who just want to advertise their businesses. We might make a custom rejection notice like this:

**Title:** Advertisement
**Message:** Based on the information you provided, it looks like you're trying to advertise your business or increase search-engine rankings. We don't allow excessive advertising or SEO manipulation on mastodon.social. Thank you for your interest.

The **Title** should show up in the drop-down menu, but *not* in the email sent to the user. The **Message** should show up in the email sent to the user, but *not* in the drop-down menu. (This allows the admins/moderators to use descriptive titles that the users might not like, and prevents the drop-down menu from being excessively long.)

The Default message should always be the first option in the drop-down menu and should be selected when the page is loaded or reloaded.

The menu might look something like this:

![Screenshot from 2019-05-03 12-01-48](https://user-images.githubusercontent.com/4139417/57150226-b5dc8f80-6d9b-11e9-9783-3590dceb356d.png)

**Comment by joyeusenoelle** [comment]
It's been suggested elsewhere that an "Other..." option be added at the bottom, which would allow the admin/moderator to write a custom one-off rejection email. This option should open a new page or bring up a modal to write the custom email, though, instead of making it part of the default interface.

(An addition to this would be to allow the moderator to then add a title and save it as a new entry in the drop-down list.)

**Comment by MirceaKitsune** [comment]
Agreed, this would be a good idea. I thought this already existed... definitely feels like one of the important missing features. I imagine some people might also take it more personally if they're rejected without a reason, without even remotely being told why they aren't being allowed in.

**Comment by trwnh** [comment]
still unsolved

**Comment by idanoo** [comment]
Just adding a comment since it's been sitting for 3 years - I think this would be huge. Especially running an instance with reasonably clear rules (e.g personal accounts only/no orgs), people don't often read the rules and it would be nice to send a reject with some preset messages, similar to the moderation warnings/messages you can send.

**Comment by mrailton** [comment]
In my opinion this is a much needed feature. Some instances (like the one I moderate on) have fairly strict joining rules which people clearly ignore. This would help us to communicate to people why their account was not approved.

**Comment by bpawel-bclub** [comment]
I would like to second this feature. It would be much appreciated! (also, while testing and rejecting my test account, I didn't get any notification that I was rejected)_

**Comment by lunalain** [comment]
Agreed this would really help

**Comment by agiletortoise** [comment]
+1. It feels really rude rejecting some accounts in cases where they did not quite understand the rules of the server.

**Comment by ShadowJonathan** [comment]
+1 with the above, where we reject users because we think they're not a great fit for the server, but like to recommend other servers.

**Comment by ThisIsMissEm** [comment]
Related to https://github.com/mastodon/mastodon/issues/12911
